www/generate_benchmark_plot.js

```diff
@@ -0,0 +1,116 @@
+// Copyright 2016 The Cockroach Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or
+// implied. See the License for the specific language governing
+// permissions and limitations under the License.
+//
+// Author: William Haack (will@cockroachlabs.com)
+
+function getTestDataFromJSON(directory, test, callback) {
+    fileName = directory + '/' + test + '.json'
+    loadJSON(fileName, callback);
+}
+
+function loadTestListFromJSON(callback) {
+    loadJSON('test_names.json', callback);
+}
+
+function getResultsFromTestData(dates, testData) {
+    var results = {
+	'N': [],
+	'A': [],
+	'B': [],
+	'M': []
+    }
+    for (var dateIndex in dates) {
+	for (key in results) {
+	    results[key].push(testData[dates[dateIndex]][key])
+	}
+    }
+    return results;
+}
+
+function getDates(testData) {
+    var dates = new Array();
+    for (var date in testData) {
+	dates.push(date);
+    }
+    dates = dates.sort(sortDates);
+    return dates
+}
+
+function initGraphs() {
+    var urlVars = getUrlVars();
+    $('#testName').html(urlVars['test']);    
+    getTestDataFromJSON(
+	urlVars['directory'], urlVars['test'], function(response) {
+	    var testData = JSON.parse(response);
+	    var dates = getDates(testData);
+	    var results = getResultsFromTestData(dates, testData);
+	    var chartTitles = {
+		'N': 'ns/op',
+		'A': 'allocs/op',
+		'B': 'B/op',
+		'M': 'MB/s'
+	    }	
+	    for (var dataType in results) {
+		var trace = {
+		    x: dates,
+		    y: results[dataType],
+		    name: urlVars['test'],
+		    mode: 'lines'
+		}
+		Plotly.newPlot(dataType + 'Plot', [trace],
+			       {title: chartTitles[dataType],
+				height: 480, width: 480});
+	    }
+	});
+}
+
+function populateCompareToSelect() {
+    loadTestListFromJSON(function(response) {
+	var dirNameToTestList = JSON.parse(response);
+	for (var dir in dirNameToTestList) {
+	    for (var testIndex in dirNameToTestList[dir]) {
```

Comment:
Again, I'd probably use `each` here. Or to append all at once, it could make sense to use `map` to generate all the elements then just append all at once (jQuery's append takes an array.)